### DISCUSSION TITLE
How to view size of files in a pull request?

### DISCUSSION THREAD

**Discussion by ashwinbalajiks** [community/community#22634] [New to GitHub]
<p>I want to see a list of files in PR and it’s size</p>

**Comment by mpboom** [comment, marked as answer]
<p>There is a tab called “Files changed” in your PR where you can see that.</p>
<p><img src="https://user-images.githubusercontent.com/19948032/181072232-a6ceceac-8a95-4b6d-82bf-18ae93949830.png" alt="Knipsel.PNG" data-base62-sha1="mubV2N01UhiiOzUqyviHVLL0n8E" width="556" height="96"></p>

**Reply by tomwheeler** [reply]
I had the same question. The provided answer provides a count of files, not the size. 

Maybe the OP was looking for the file count, but I was looking for the delta (in # of bytes) that the PR represents. In other words, how much bigger or smaller would the codebase become as a result of merging it.

**Reply by mpboom** [reply]
@tomwheeler I don't think GitHub has any way to provide for file size specifically. Technically, a "PR" really just is a pointer to a set of two commits. That's very tiny. You want to know how much the HEAD increases in bytes or something? What is the use case? You could probably build an action or something that provides you that information.

**Reply by tomwheeler** [reply]
Yesterday I received a PR for a feature request whose implementation was much more complicated than I expected. It added more than 2,000 files to the repository. Since that repository is one that many users (not just developers) need to clone, a major increase in size would be one reason to reject the request. I can quantify the count of files through git, but was looking for a delta for the increase in size (# of bytes). I can solve that through a sequence of UNIX commands easily enough, but didn't know whether git provided support for it directly.

**Reply by yo1995** [reply]
It could be a helpful indicator on whether to squash the commits or not. Say if some big chunk of changes or binary files were committed but later taken out. Having a total bytes commit metric can help decide whether to keep the history, or squash them when they aren't worth the extra space.
